### ISSUE TITLE
vis lässt sich seit backup nicht installieren

### ISSUE THREAD

**Issue by Ptikbrn** [ioBroker/ioBroker.vis#440]
log Eintrag sieht so aus ...

159 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/zigbee-herdsman/node_modules/fsevents):
160 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
161 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
161 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
161 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
161 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64
162 verbose stack Error: iobroker.vis@1.4.7 install: `node main.js --install`
162 verbose stack Exit status 6
162 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
162 verbose stack     at EventEmitter.emit (events.js:400:28)
162 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
162 verbose stack     at ChildProcess.emit (events.js:400:28)
162 verbose stack     at maybeClose (internal/child_process.js:1058:16)
162 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)
163 verbose pkgid iobroker.vis@1.4.7
164 verbose cwd /opt/iobroker
165 verbose Linux 5.10.0-10-amd64
166 verbose argv "/usr/bin/node" "/usr/bin/npm" "install" "iobroker.vis@1.4.7" "--loglevel" "error" "--prefix" "/opt/iobroker"
167 verbose node v14.18.2
168 verbose npm  v6.14.15
169 error code ELIFECYCLE
170 error errno 6
171 error iobroker.vis@1.4.7 install: `node main.js --install`
171 error Exit status 6
172 error Failed at the iobroker.vis@1.4.7 install script.
172 error This is probably not a problem with npm. There is likely additional logging output above.
173 verbose exit [ 6, true ]

vllt kann irgendwer damit was anfangen bzw helfen ?

**Comment by Apollon77** [comment]
Please try to run manually 

`
cd /opt/iobroker
npm i iobroker.vis@1.4.7
`

Is there more output?

**Comment by Ptikbrn** [comment]
manuell hatte ich auch versucht und endet mit genau dem selben Fehler ...... ich weiß mir aber auch keinen Rat mehr wieso es jetzt auf einmal nicht mehr geht ...... habe nur im gleichen Atemzug beim System neu aufsetzen von Debian 10 auf 11 geupdated

**Comment by Ptikbrn** [comment]
> iobroker.vis@1.4.7 install /opt/iobroker/node_modules/iobroker.vis
> node main.js --install

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: osx-temperature-sensor@1.0.7 (node_modules/osx-temperature-sensor):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for osx-temperature-sensor@1.0.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/zigbee-herdsman-converters/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/zigbee-herdsman-converters/node_modules/zigbee-herdsman/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/zigbee-herdsman/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 6
npm ERR! iobroker.vis@1.4.7 install: `node main.js --install`
npm ERR! Exit status 6
npm ERR! 
npm ERR! Failed at the iobroker.vis@1.4.7 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/iobroker/.npm/_logs/2022-01-07T10_16_51_660Z-debug.log

das wird mir bei manuell ausführen angezeigt

**Comment by Ptikbrn** [comment]
nach hin und her und gefühlten 5x fixer habe ich jetzt einen neuen Fehler ..... als ersten spuckt er aus keine Lizenz gefunden , aber die kann ich doch erst eingeben und überprüfen , wenn vis installiert ist aber dazu kommt es nicht Mal

**Comment by Ptikbrn** [comment]
2022-01-07 12:29:15.334 | error | iobroker host.iobroker Cannot install iobroker.vis@1.4.7: 6
-- | -- | --
vis.0 | 2022-01-07 12:29:12.473 | error | Not exists
vis.0 | 2022-01-07 12:29:12.473 | error | Error: Not exists at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2335:17) at ObjectsInRedisClient._readFile (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:745:26) at Immediate.<anonymous> (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:792:29) at processImmediate (internal/timers.js:466:21)
vis.0 | 2022-01-07 12:29:12.472 | error | unhandled promise rejection: Not exists
vis.0 | 2022-01-07 12:29:12.471 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
vis.0 | 2022-01-07 12:29:12.452 | error | No license found for vis. Please get one on https://iobroker.net !

**Comment by Ptikbrn** [comment]
iobroker host.iobroker Cannot install iobroker.vis@1.4.7: 6
2022-01-07 12:29:12.473 Not exists
2022-01-07 12:29:12.473 Error: Not exists at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2335:17) at ObjectsInRedisClient._readFile (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:745:26) at Immediate.<anonymous> (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:792:29) at processImmediate (internal/timers.js:466:21)
2022-01-07 12:29:12.472 promise rejection: Not exists
2022-01-07 12:29:12.471 Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2022-01-07 12:29:12.452 No license found for vis. Please get one on https://iobroker.net !

**Comment by Ptikbrn** [comment]
und ja ich habe eine Lizenz bei iobroker.net und der Cloud Adapter läuft auch und ist verbunden

**Comment by Apollon77** [comment]
Kannst du die 1.4.6 installieren?

**Comment by Apollon77** [comment]
oder 1.4.5?

**Comment by Apollon77** [comment]
oder 1.4.4? Versuch mal zurück. Ich hab eine befürchtung
Ich gehe davon aus das Du keine Instanz hast wo die vis Lizenz schon eingetragen ist?

**Comment by Ptikbrn** [comment]
aktuell läuft gar keine Instanz mit vis ich bekomme ja über ioBroker web ui
kein vis installiert .....

Ingo Fischer ***@***.***> schrieb am Fr., 7. Jan. 2022, 13:17:

**Comment by Apollon77** [comment]
Ja dann mach malo btte ausnahmsweise wie oben beschrieben direkt mit dem "npm i ..." befehlt und versuch mal ob die 1.4.6 geht ... wenn nicht versuch die 1.4.5 ... kannst später updaten ... aber soköpnntest du mir eine befürchtung verifizieren

Wenn ein upload geklappt hat dann noch ein "iob upload vis" hinterher

**Comment by Ptikbrn** [comment]
von 1.4.4 bis 1.4.6 probiert alles spucken das aus

> ***@***.*** install /home/patrick/node_modules/iobroker.vis
> node main.js --install

npm WARN saveError ENOENT: no such file or directory, open
'/home/patrick/packag
e.json'
npm WARN enoent ENOENT: no such file or directory, open
'/home/patrick/package.j
son'
npm WARN patrick No description
npm WARN patrick No repository field.
npm WARN patrick No README data
npm WARN patrick No license field.

+ ***@***.***
updated 1 package and audited 34 packages in 0.912s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Ingo Fischer ***@***.***> schrieb am Fr., 7. Jan. 2022, 13:25:

**Comment by Apollon77** [comment]
ääähhmm ... naja machs mal in /opt/iobroker und nicht in deinem home dir

**Comment by Ptikbrn** [comment]
so schnell kann es gehen ..... ;-)

hat geklappt , habe mit npm die vis 1.4.6 laden können habe dann den upload
Befehl gemacht könnte dann unter Adapter sehen daß er sein war habe dann
Update auf 1.4.7 gemacht und dann auf das plus und somit als Instanz hinzu
gefügt .... api Key eingeben und es läuft wieder



vielen vielen Dank ☺️

Ingo Fischer ***@***.***> schrieb am Fr., 7. Jan. 2022, 13:48:

**Comment by Apollon77** [comment]
@GermanBluefox So assumotion is that vis 1.4.7 is BROKEN for all new installations? Could this be?

**Comment by Ptikbrn** [comment]
würde ich jetzt nicht sagen, vor dem System Crash gestern bei mir leid ja
wochenlang vis in Version 1.4.7 ..... das hat irgendwas mir backitup und
dem Backup einspielen zu tun

Ingo Fischer ***@***.***> schrieb am Fr., 7. Jan. 2022, 14:29:

**Comment by D3ltorohd** [comment]
Ich hänge mich hier mal mit an. Ich habe auch ein neues System aufgesetzt und danach mit Backitup alles wieder eingespielt, alles lief durch bis VIS Installation,

> host.ioBroker
	2022-01-09 14:34:39.434	info	Restart adapter system.adapter.backitup.0 because enabled
host.ioBroker
	2022-01-09 14:34:39.434	error	instance system.adapter.backitup.0 terminated with code 1 (JS_CONTROLLER_STOPPED)
host.ioBroker
	2022-01-09 14:34:39.434	error	Caught by controller[0]: at Pipe.<anonymous> (net.js:686:12)
host.ioBroker
	2022-01-09 14:34:39.434	error	Caught by controller[0]: at Socket.emit (domain.js:475:12)
host.ioBroker
	2022-01-09 14:34:39.434	error	Caught by controller[0]: at Socket.emit (events.js:400:28)
host.ioBroker
	2022-01-09 14:34:39.434	error	Caught by controller[0]: at Socket.<anonymous> (internal/child_process.js:443:11)
host.ioBroker
	2022-01-09 14:34:39.433	error	Caught by controller[0]: at maybeClose (internal/child_process.js:1058:16)
host.ioBroker
	2022-01-09 14:34:39.433	error	Caught by controller[0]: at ChildProcess.emit (domain.js:475:12)
host.ioBroker
	2022-01-09 14:34:39.433	error	Caught by controller[0]: at ChildProcess.emit (events.js:400:28)
host.ioBroker
	2022-01-09 14:34:39.433	error	Caught by controller[0]: at ChildProcess.exithandler (child_process.js:390:5)
host.ioBroker
	2022-01-09 14:34:39.433	error	Caught by controller[0]: at /opt/iobroker/node_modules/iobroker.backitup/lib/list/cifs.js:169:49
host.ioBroker
	2022-01-09 14:34:39.433	error	Caught by controller[0]: at new RegExp (<anonymous>)
host.ioBroker
	2022-01-09 14:34:39.432	error	Caught by controller[0]: SyntaxError: Invalid regular expression: /'6o:?'d vfx(v'/: Unterminated group
backitup.0
	2022-01-09 14:34:39.341	warn	Terminated (UNCAUGHT_EXCEPTION): Without reason
backitup.0
	2022-01-09 14:34:39.341	info	terminating
backitup.0
	2022-01-09 14:34:39.340	info	cleaned everything up...
backitup.0
	2022-01-09 14:34:39.333	error	Invalid regular expression: /'6o:?'d vfx(v'/: Unterminated group
backitup.0
	2022-01-09 14:34:39.333	error	SyntaxError: Invalid regular expression: /'6o:?'d vfx(v'/: Unterminated group at new RegExp (<anonymous>) at /opt/iobroker/node_modules/iobroker.backitup/lib/list/cifs.js:169:49 at ChildProcess.exithandler (child_process.js:390:5) at ChildProcess.emit (events.js:400:28) at ChildProcess.emit (domain.js:475:12) at maybeClose (internal/child_process.js:1058:16) at Socket.<anonymous> (internal/child_process.js:443:11) at Socket.emit (events.js:400:28) at Socket.emit (domain.js:475:12) at Pipe.<anonymous> (net.js:686:12)
backitup.0
	2022-01-09 14:34:39.333	error	uncaught exception: Invalid regular expression: /'6o:?'d vfx(v'/: Unterminated group
backitup.0
	2022-01-09 14:34:39.279	info	[iobroker] backup was activated at 02:40 every 1 day(s)
backitup.0
	2022-01-09 14:34:39.252	info	starting. Version 2.2.2 in /opt/iobroker/node_modules/iobroker.backitup, node: v14.18.2, js-controller: 3.3.22
host.ioBroker
	2022-01-09 14:34:38.443	info	instance system.adapter.backitup.0 started with pid 18774
host.ioBroker
	2022-01-09 14:34:24.968	error	Cannot download and install adapter "vis@1.4.7". To retry it disable/enable the adapter or restart host. Also check the error messages in the log!
host.ioBroker
	2022-01-09 14:34:23.967	info	iobroker npm-install: exit 25
host.ioBroker
	2022-01-09 14:34:22.956	error	iobroker npm-install: host.ioBroker Cannot install iobroker.vis@1.4.7: 6
host.ioBroker
	2022-01-09 14:34:22.936	info	iobroker npm-install: npm ERR! A complete log of this run can be found in:npm ERR! /home/iobroker/.npm/_logs/2022-01-09T14_34_22_924Z-debug.log
host.ioBroker
	2022-01-09 14:34:22.935	info	iobroker npm-install:
host.ioBroker
	2022-01-09 14:34:22.919	info	iobroker npm-install: npm ERR! Exit status 6npm ERR! npm ERR! Failed at the iobroker.vis@1.4.7 install script.npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
host.ioBroker
	2022-01-09 14:34:22.919	info	iobroker npm-install: npm ERR! iobroker.vis@1.4.7 install: `node main.js --install`
host.ioBroker
	2022-01-09 14:34:22.914	info	iobroker npm-install: npm ERR! errno 6
host.ioBroker
	2022-01-09 14:34:22.912	info	iobroker npm-install: npm ERR! code ELIFECYCLE
vis.0
	2022-01-09 14:34:20.687	warn	Terminated (UNCAUGHT_EXCEPTION): Without reason
vis.0
	2022-01-09 14:34:20.686	info	terminating
vis.0
	2022-01-09 14:34:20.182	error	Not exists
vis.0
	2022-01-09 14:34:20.182	error	Error: Not exists at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2335:17) at ObjectsInRedisClient._readFile (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:745:26) at Immediate.<anonymous> (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:792:29) at processImmediate (internal/timers.js:466:21)
vis.0
	2022-01-09 14:34:20.181	error	unhandled promise rejection: Not exists
vis.0
	2022-01-09 14:34:20.181	error	Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
vis.0
	2022-01-09 14:34:20.058	info	vis license is OK.
vis.0
	2022-01-09 14:34:19.777	info	starting. Version 1.4.7 in /opt/iobroker/node_modules/iobroker.vis, node: v14.18.2, js-controller: 3.3.22
host.ioBroker
	2022-01-09 14:34:08.426	info	Restart adapter system.adapter.backitup.0 because enabled

**Comment by Apollon77** [comment]
Das ist abe rirgendwie ein Fehler von backit up``

```
 2022-01-09 14:34:39.333 error SyntaxError: Invalid regular expression: /'6o:?'d vfx(v'/: Unterminated group at new RegExp () at /opt/iobroker/node_modules/iobroker.backitup/lib/list/cifs.js:169:49 at ChildProcess.exithandler (child_process.js:390:5) at ChildProcess.emit (events.js:400:28) at ChildProcess.emit (domain.js:475:12) at maybeClose (internal/child_process.js:1058:16) at Socket. (internal/child_process.js:443:11) at Socket.emit (events.js:400:28) at Socket.emit (domain.js:475:12) at Pipe. (net.js:686:12)
 ```
 
 Wäre also eher ein Issue für dort!

**Comment by frankjoke** [comment]
Had same problem today after restore of backup on a fresh iobroker instance (Raspi bullseye). Could not install vis (in restore and manually) because of above reason. could also not uninstall adapter!
What I did was to `iob install vis@1.4.5` which worked then and after that I could also upgrade to latest 1.4.7 version!
But my restore vis is of course missing (: Will see now how I can restore vis pages from backup!

**Comment by Apollon77** [comment]
Und ja weil die Konfig fehl tgibts das gleiche issue wie oben: Mit 1.4.6 sollte es gehen ... aber dein issue ist eher das unvolstsändige restore oder so von backitup

**Comment by Apollon77** [comment]
I will re-set the latest verison to 1.4.6 of vis ... so with next repo update the issue should be gone ... vis 1.4.7 works as update but have issues on new installs without config

**Comment by D3ltorohd** [comment]
Ok ich probier den Restore gleich noch mal. Spiele grad das Backup vom Container ein.

**Comment by Ptikbrn** [comment]
weiter oben gibts es eine tolle Schritt für Schritt die bei mir geholfen
hat .....

D3ltorohd ***@***.***> schrieb am So., 9. Jan. 2022, 17:05:

**Comment by Apollon77** [comment]
@D3ltorohd zu  dem backitup Thema habe ich die Info schon weitergegeben aber gern da noch ein issue anlegen. Das liegt an der Klammer in deinem passwort

**Comment by D3ltorohd** [comment]
Welches Passwort @Apollon77 Das sieht man jetzt in der Log, hätte ich wohl vorher raus nehmen sollen.

**Comment by Apollon77** [comment]
Kannst editieren ... und weisste wo es knallt? GEnau an der code stelle wo backitup versucht das passwort für log ausgaben durch *** zu ersetzen *gg*

**Comment by D3ltorohd** [comment]
Sorry ich find da nichts im Log, indem was ich oben gepostet habe ? Was soll das für ein Passwort sein ? Vom admin, von der Cifs Freigabe ? Ich habe da nirgends Klammern drin.

**Comment by simatec** [comment]
@D3ltorohd Es geht um dein Passwort zur CIFS Freigabe. Die ganzen Sonderzeichen machen unter Unix Probleme
Habe es gerade hier auf meinem Testsystem nachgestellt ... Wähle mal ein Passwort ohne Sonderzeichen und es sollte nicht allzu lang sein

**Comment by D3ltorohd** [comment]
@simatec Wie seht ihr das Passwort, oder ist das die Schlussfolgerung aus der Fehlerlog, das ich dort Sonderzeichen verwende ?

Ok, dann muss ich schauen wie ich das anstelle. Müsste ich auf dem Linux System einen neuen Benutzer und ein neues Passwort anlegen, oder kann ich für die Freigabe ein anderes Passwort vergeben, bin in Linux nicht so bewandert. Zum speichern des Backups gibt es aber keine Probleme. Das ist ja ein Linux Server auf dem das Backup liegt.

**Comment by Apollon77** [comment]
Nee wrte bis @simatec es gefixt hat :-)

**Comment by simatec** [comment]
Der Fehler in Backitup ist mit Version 2.2.3 gefixt

**Comment by JogiBarbarski** [comment]
Ich danke euch. Ich hatte das gleiche Problem.
Raspberry Pi 4, 8GB
ioBroker.admin 5.3.3
Node.js: v14.19.1
NPM: 6.14.16
js-controller
JS controller Installiert: 4.0.21
OS: linux
Latest 
Nach backup war die VIS weg. Konnte den Adapter nicht löschen und nicht neu installieren. Es ging nichts. Dank eurer Anleitung oben, über die Shell habe ich alles wieder hinbekommen. Herzlichen Dank. In so einer VIS steckt eine menge Herzblut drin. 
Falls noch jemand das Problem hat und nicht so fit ist, 
ich habe über SSH "  /opt/iobroker $ " "  iob install vis@1.4.5 "durchgeführt, danach " iobroker update " und anschließend  " iobroker upgrade " durchgeführt . 
Jetzt ist alles wieder da. 
Schöne Grüße.

**Comment by kBrausew** [comment]
Es ist immer noch aktuell... 
Hab es auf der Konsole mit
`npm i iobroker.vis@1.4.6`
beheben können.

**Comment by Trrw** [comment]
Bei mir das gleiche Problem. 

Ich bekomme in der Konsole folgenden Fehler:

```
/opt/iobroker# npm i iobroker.vis@1.4.6

> iobroker.vis@1.4.6 install /opt/iobroker/node_modules/iobroker.vis
> node main.js --install

npm WARN http-cookie-agent@1.0.6 requires a peer of tough-cookie@^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: osx-temperature-sensor@1.0.7 (node_modules/osx-temperature-sensor):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for osx-temperature-sensor@1.0.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

+ iobroker.vis@1.4.6
added 1 package from 2 contributors in 42.399s

128 packages are looking for funding
  run `npm fund` for details
```

**Comment by Apollon77** [comment]
Da st aber kein Fehler ... nurr Warnungen ... oder ws genau ist das ossue? Das ist alles erfolgreich

**Comment by tobasium** [comment]
Also ich hab heute auch ein Backup eingespielt und es hat nicht ganz geklappt:

<html>
<body>


vis.0 | 2022-07-18 17:14:10.936 | error | Not exists
-- | -- | -- | --
vis.0 | 2022-07-18 17:14:10.935 | error | Error: Not exists at Object.maybeCallbackWithError (/opt/iobroker/node_modules/@iobroker/js-controller-common/lib/common/tools.js:2973:17) at ObjectsInRedisClient._readFile (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:1043:26) at Immediate.<anonymous> (/opt/iobroker/node_modules/@iobroker/db-objects-redis/lib/objects/objectsInRedisClient.js:1092:29) at processImmediate (node:internal/timers:468:21)
vis.0 | 2022-07-18 17:14:10.932 | error | unhandled promise rejection: Not exists
vis.0 | 2022-07-18 17:14:10.931 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().


</body>
</html>

**Comment by xmace** [comment]
Hi,
bei mir auch dasselbe. VIS lässt sich nach einem restore nicht mehr installieren.

**Comment by franky341** [comment]
hier auch dasselbe Problem

**Comment by RkcCorian** [comment]
Hab ein Restore gemacht (als Update von Container v6.0.0 auf v7.0.1. Hab das gleiche Problem, dass VIS sich nicht installieren lässt. Auch nachdem ich es über ```iob fix```, sowie ```cd /opt/iobroker``` und ```npm i iobroker.vis@1.4.15``` installiert habe, lief es danach nicht (konnte nicht auf die Web-Oberfläche)

**Comment by klante0208** [comment]
Hallo Leute,
ich habe da mal einen anderen Ansatz verfolgt.
1. npm Version auf übereinstimmung mit der des Backups prüfen.
2. den .vis-Adapter vor dem Backup installieren und lizensieren
zu 1. : die backup.tar.gz entpacken. Im Verzeichnis backup die Datei backup.json öffnen. Nach "npmCurrent" suchen. Das gibt in der Regel vier Treffer. In dem Block mit den Parametern  "val", "ack", "ts" usw. steht hinter "val" die Versionsnummer, die zum Zeitpunkt des Backups installiert war. Diese sollte mit der des neuen Systems übereinstimmen. 
Das bekommt man mit `npm -v` heraus. Sollte die Version nicht übereinstimmen, mit 
`sudo npm install -g  npm@versionsnummer` down-/upgraden. Danach den Iobroker wie gewohnt installieren und grundkonfigurieren.
zu 2. : den vis-Adapter installieren und lizensieren. 
Nun kann man den Iobroker aus dem Backup wiederherstellen. 
Nach einigen vergeblichen Versuchen, siehe oben, hat nur das bei mir geklappt.
Viel Erfolg

**Comment by Apollon77** [comment]
Die npm Version sollte hier keinerlei Auswirkung haben

**Comment by BassT23** [comment]
Hatte heute das selbe problem.

Backup wurde mit mpn 16 gemacht.
iob neu aufgesetzt mit npm 18.
Backup wieder her gestellt -> vis lässt sich nicht installieren.

Lösung:
iob deinstallieren:
`cp -r /opt/iobroker/iobroker-data ~/iobroker-data`
`sudo rm -r /opt/iobroker/ `

iob neu installieren
erster start -> web und vis installieren
backup wiederherstellen

im log steht zwar, dass vis deinstalliert wird, aber nach der backup wiederherstellung wurde vis wie gewünsch installiert.

Hoffe das hilft auch anderen ;)

**Comment by Apollon77** [comment]
@BassT23 Alkao das ist aber ein krasser weg. Ja er geht, aber meine empfehlung wäre er nicht

**Comment by RkcCorian** [comment]
I also had issues and after restore VIS cannot be installed / get going.

Procedure: I have buanet container v6.0.0 running since months and realized that there is a new v7.0.1. I created a backup (Backitup) and restored by putting the backup in the new iobroker folder and created a new container as usual. When it comes to the installation of adapters it fails for the VIS. This was not a problem with v6.0.0 before.

In sum the following I did to workaround it. It has to be done AFTER restored everything AND BEFORE restarting for the adapter installation:
a. „pkill -u iobroker“ executed
b. „cd /opt/iobroker“ executed (only in case not already in the folder)
c. „npm i iobroker.vis@1.4.15” (choose your version) executed
d. „iobroker upload vis“ executed

I hope this will be fixed (or workaround will be in place) for future versions. It is really anoying and very bad when restoring or upgrading.

Shall this issue be reopened?!

**Comment by BassT23** [comment]
> a. „pkill -u iobroker“ executed
> b. „cd /opt/iobroker“ executed (only in case not already in the folder)
> c. „npm i iobroker.vis@1.4.15” (choose your version) executed
> d. „iobroker upload vis“ executed

Confirm, work for me.
I use an Linux install so, for me in a. I use "iob stop"

**Comment by RkcCorian** [comment]
@GermanBluefox : Anyhow that the workaround mentioned by me is working... is it possible to fix this for the future to get rid of this ugly workaround?! Many thanks!

**Comment by RkcCorian** [comment]
@GermanBluefox : Any update here? Many thanks!

**Comment by RkcCorian** [comment]
@GermanBluefox : Any update here? Many thanks!
@Apollon77 : Do you know something? Many thanks!

**Comment by CampingSchlumpf** [comment]
**Annoying, but easy way to fix it:**
1. Install "npm i iobroker.vis@1.4.6" manually
2. Upgrade to 1.4.15 via the WebGui

**Comment by RkcCorian** [comment]
@CampingSchlumpf 
If I remember right I tried this but failed.
Still this shall be fixed to reduce the hurdle of using ioBroker, esp. for user with lesser background which would like to "simply" use ioBroker. And Backup & Restore is an important point in my opinion.

Therefore I'm hoping for a bugfix in the near future...

**Comment by Bodengriller** [comment]
Kann bestätigen, dass das Problem noch besteht.
über die Konsole 
```
cd /opt/iobroker 
npm i iobroker.vis@1.4.6
```
und anschließendes Update übers WebUI klappt's dann.

**Comment by garfield2412** [comment]
> Kann bestätigen, dass das Problem noch besteht. über die Konsole
> 
> ```
> cd /opt/iobroker 
> npm i iobroker.vis@1.4.6
> ```
> 
> und anschließendes Update übers WebUI klappt's dann.

selbe ausgangslage hier... Backup eingespielt, geht, aber VIS lässt sich nicht installieren/Updaten/runnen...
Deine Lösung angewandt - geht, aber mit dem unterschied das ich nach der installation die 1.4.6 einmal starten musste (objekt-Upload) und dann die 1.4.15 über die GUI drüber knallen konnte.

Als folge danach ließen sich auch andere Adapter wieder updaten (vermutlich mit VIS Abhängigkeiten)

DANKE!

**Comment by RkcCorian** [comment]
@GermanBluefox , @Apollon77 : Wird das überhaupt noch gefixt oder erst für VIS2.0? Vielen Dank für die Rückmeldung!

**Comment by Apollon77** [comment]
Geplant ist dies noch vor Vis 2.0 zu fixen.

**Comment by JayVee2** [comment]
Bin gerade nach einem vollständigen System Restore in dieselbe Falle getappt.
Workaround geht wie oben beschrieben.

**Comment by ChrisBGL** [comment]
Problem besteht immer noch! Restore ohne den Workaround nicht möglich.

**Comment by Percy2Live** [comment]
kann ich bestätigen, Problem ist noch präsent

**Comment by Grizzelbee** [comment]
The Javascript-Adapter 7.x.x makes this issue more complicated.
With Javascript-Adapter 7.0.3 today I received:

```
server: /opt/iobroker$ npm install iobroker.vis@1.4.6
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /opt/iobroker/node_modules/iobroker.javascript/node_modules/typescript
npm ERR! dest /opt/iobroker/node_modules/iobroker.javascript/node_modules/.typescript-EcwXdkx5
npm ERR! errno -39
npm ERR! ENOTEMPTY: directory not empty, rename '/opt/iobroker/node_modules/iobroker.javascript/node_modules/typescript' -> '/opt/iobroker/node_modules/iobroker.javascript/node_modules/.typescript-EcwXdkx5'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2023-06-20T07_15_01_978Z-debug-0.log
```

I resolved it by downgrading javascript to 6.0.0 and updating all via WebGUI after that.

So I used:
```
server: /opt/iobroker/$ npm install iobroker.javascript@6.0.0
server: /opt/iobroker/$ npm install iobroker.vis@1.4.6
```
to fix this issue.

**Comment by Apollon77** [comment]
Der obige Fehler lässt sich an sich mit "iob fix" lösen!
Siehe auch https://forum.iobroker.net/topic/57337/fehler-25-217-oder-enotempty-bei-adapter-install-update
